Source/WebCore/dom/FragmentDirectiveGenerator.cpp

```diff
@@ -102,38 +102,49 @@ void FragmentDirectiveGenerator::generateFragmentDirective(const SimpleRange& te
     String textDirectivePrefix = ":~:text="_s;
 
     auto url = textFragmentRange.startContainer().document().url();
-
     auto textFromRange = createLiveRange(textFragmentRange)->toString();
 
     VisiblePosition visibleEndPosition = VisiblePosition(Position(textFragmentRange.protectedEndContainer(), textFragmentRange.endOffset(), Position::PositionIsOffsetInAnchor));
     VisiblePosition visibleStartPosition = VisiblePosition(Position(textFragmentRange.protectedStartContainer(), textFragmentRange.startOffset(), Position::PositionIsOffsetInAnchor));
 
-    if (textFromRange.length() >= maxCharacters) {
-        String startText = percentEncodeFragmentDirectiveSpecialCharacters(nextWordsFromPosition(3, visibleStartPosition));
-        String endText = percentEncodeFragmentDirectiveSpecialCharacters(previousWordsFromPosition(3, visibleEndPosition));
-
-        url.setFragmentIdentifier(makeString(textDirectivePrefix, startText, ',', endText));
-    }
-    if (textFromRange.length() < maxCharacters && textFromRange.length() > minCharacter) {
-        StringBuilder textDirectiveBuilder;
-        textDirectiveBuilder.append(textDirectivePrefix);
-        textDirectiveBuilder.append(percentEncodeFragmentDirectiveSpecialCharacters(textFromRange));
-        url.setFragmentIdentifier(StringView(textDirectiveBuilder));
-    }
-    if (textFromRange.length() <= minCharacter) {
-        String prefix = percentEncodeFragmentDirectiveSpecialCharacters(previousWordsFromPosition(3, visibleStartPosition));
-        String suffix = percentEncodeFragmentDirectiveSpecialCharacters(nextWordsFromPosition(3, visibleEndPosition));
-
-        url.setFragmentIdentifier(makeString(textDirectivePrefix,
-            percentEncodeFragmentDirectiveSpecialCharacters(prefix), "-,"_s,
-            percentEncodeFragmentDirectiveSpecialCharacters(textFromRange), ",-"_s,
-            percentEncodeFragmentDirectiveSpecialCharacters(suffix)));
+    std::optional<String> prefix;
+    std::optional<String> startText;
+    std::optional<String> endText;
+    std::optional<String> suffix;
+
+    auto encodeComponent = [] (const String& component) -> std::optional<String> {
+        auto encodedComponent = percentEncodeFragmentDirectiveSpecialCharacters(component);
+        if (encodedComponent.isEmpty())
+            return std::nullopt;
+        return encodedComponent;
+    };
+
+    if (textFromRange.length() >= maximumInlineStringLength) {
+        startText = encodeComponent(nextWordsFromPosition(3, visibleStartPosition));
```

Comment:
should the `3` be extracted to a named constant ?

Reply:
3 is also elsewhere in the file, so change those to this constant as well, please!